wolfcrypt/src/asn.c

```diff
@@ -16727,70 +16726,75 @@ static int DecodeSingleResponse(byte* source,
         return ASN_PARSE_E;
 
     /* Wrapper around the CertID */
+    certIdIdx = idx;
     if (GetSequence(source, &idx, &length, size) < 0)
         return ASN_PARSE_E;
-    /* Skip the hash algorithm */
-    if (GetAlgoId(source, &idx, &oid, oidIgnoreType, size) < 0)
-        return ASN_PARSE_E;
+    single->rawCertId = source + certIdIdx;
+    /* Hash algorithm */
+    ret = GetAlgoId(source, &idx, &oid, oidIgnoreType, size);
+    if (ret < 0)
+        return ret;
+    single->hashAlgoOID = oid;
     /* Save reference to the hash of CN */
     ret = GetOctetString(source, &idx, &length, size);
     if (ret < 0)
         return ret;
-    resp->issuerHash = source + idx;
+    XMEMCPY(single->issuerHash, source + idx, OCSP_DIGEST_SIZE);
```

Comment:
What if length is < OCSP_DIGEST_SIZE?  Don't we over read in that case?

Reply:
Agreed. Will change OCSP_DIGEST_SIZE to length.